<span style=" font-family: Babas; font-size: 2em;">Michelle Ukiwe  - Homework 2


In [2]:
import math as math

#### Altitude of a satellite:
A satellite is to be launched into a circular orbit around the Earth so that it orbits the planet once every _T_ seconds. Treating the Earth as a perfect sphere with its center of mass in the middle (which is only approximately correct), it can shown with classical mechanics that the altitude _h_ above the Earth’s surface must be 

\begin{equation}
h = (\frac{GMT^2}{4\pi^2})^\frac{1}{3} - R
\end{equation}





where \begin{equation}
G = 6.67 × 10^{−11} m^3 kg^{-1} s^{−2} 
\end{equation}

is Newton’s gravitational constant, 
\begin{equation}
M = 5.97 × 10^{24} kg
\end{equation}is the Earth's mass, and 

\begin{equation}
R = 6371  km 
\end{equation}
is its radius.

__Part (a)__\
Write a program that asks the user to enter the desired value of _T_ and then calculates and prints out the corresponding altitude in meters

In [86]:
#defining a function that takes the time and outputs altitude
def altitude(T):
    #gravitational constant
    grav = 6.67 * (10**(-11))
    #the mass of the earth
    mass = 5.97 * (10**24)
    #the radius of the earth converted to meters
    rad = 6371*1000
    #the numerator of the altitude equation
    numerator = (grav * mass * (T**2))
    #the denominator of the equation
    denominator = 4*((math.pi)**2)
    #calculation of altitude
    altitude = ((numerator/denominator)**(1/3)) - rad
    #return the time and altitude
    print("the altitude at " + str(T) + " seconds is " + str(altitude) + " meters")
#input for time
T = float(input("Enter in a value for T time in seconds: "))

altitude(T)

Enter in a value for T time in seconfds:  3


the altitude at 3.0 seconds is -6326057.111709947 meters


__Part (b)__\
Use your program to calculate the altitudes of satellites that orbit the Earth once a day (so-called “geosynchronous” orbit), once every 90 minutes, and once every 45 minutes. What do you conclude from the last of these calculations?

In [4]:
#86400 minutes in a day
daily = 86400
#5400 minutes in a day
ninety = 5400
#2700 seconds
fortyfive = 2700

In [83]:
altitude(daily)

the altitude at 86400 seconds is 35855910.17617497 meters


In [84]:
altitude(ninety)

the altitude at 5400 seconds is 279321.62537285965 meters


In [85]:
altitude(fortyfive)

the altitude at 2700 seconds is -2181559.8978108233 meters


__Answers to part b in markdown__ \
daily orbit: the altitude at 86400 seconds is 35855910.17617497 meters \
90-minute orbit: the altitude at 5400 seconds is 279321.62537285965 meters \
45-minute orbit: the altitude at 2700 seconds is -2181559.8978108233 meters

From this, it is shown that a satellite's altitude should be closer to Earth if the orbit is to be completed in a shorter amount of time and vice-versa

__Part (c)__\
Technically a geosynchronous satellite is one that orbits the Earth once per _sidereal day_, which is 23.93 hours not 24 hours. Why is this? How much difference will it make to the altitude of the satellite?

In [9]:
sidereal = 23.93*60*60
#variable named sidereal that holds seconds in a sidreal day
sidereal

86148.0

In [87]:
altitude(sidereal)

the altitude at 86148.0 seconds is 35773762.329895645 meters


In [89]:
#altitude for a sidreal day
alt2 = 35773762.329895645
#altitude for a 24hr day
alt1 = 35855910.17617497
#returns difference
diff = alt1 - alt2
diff

82147.8462793231

__Answer to part c in markdown__

geosynchronous satellites orbit the earth once per sidereal day to mimic the Earth rotation on its axis. The difference in altitude for a satellite that orbits the Earth every sidereal day versus every twenty-four hours is approximately 82147.85 meters

#### Special relativity:
A spaceship travels from Earth in a straight line at a relativistic speed _v_ to another planet _x_ light-years away. Time dilation is a special relativistic effect. An observer with some speed _v_ relative to another observer will experience time progressing at a slower rate. Let ∆t′ be the time interval for the moving frame, and ∆t be the rest frame. They are related through 


\begin{equation}
∆t^′ = ∆t\sqrt{(1-\frac{v^2}{c^2})}
\end{equation}

Write a program to ask the user for the value of _x_ and the speed _v_ as a fraction of the speed of light _c_. Print out the time in years that the spaceship takes to reach its destination 

__Part (a)__ \
in the rest frame of an observer on Earth 

In [105]:
#defining a function in which the rest frame is from the perspective of an observer on earth
def restframe():
    #asking user to input the distance from earth
    distance = float(input("how many light years away is the planet from Earth?: "))
    #asking the user the speed of the spaceship
    speed = float(input("enter the speed as a fraction of the speed of light: "))
    #calculating how long it will take the ship to reach its destination from a viewer on earth's perspective
    propertime = distance / speed
    #print the result
    print("it will take the spaceship " + str(propertime) + " years to reach it's destination from the perspective of earth")


__Part (b)__\
as perceived by a passenger on board the ship. 

In [106]:
#defining a function that returns how long it takes for the viewer to reach their destination with the effects of time dilation taken into account
def spaceshipframe():
    #asking distance and speed
    distance = float(input("how many light years away is the planet from Earth?: "))
    speed = float(input("enter the speed as a fraction of the speed of light: "))
    #calculating delta t, or proper time
    propertime = distance / speed
    #calculating delta t prime. c is 1, so its not included here
    tprime = propertime *(math.sqrt(1 - (speed**2)))
    #returning result
    print("it will take the spaceship " + str(tprime) + " years to reach it's destination from the perspective of people on the spaceship")


Use your program to calculate the answers for a planet 10 light-years away with v = 0.99c.

In [102]:
restframe()

how many light years away is the planet from Earth?:  10
enter the speed as a fraction of the speed of light:  0.99


it will take the spaceship 10.1010101010101 years to reach it's destination from the perspective of earth


In [103]:
spaceshipframe()

how many light years away is the planet from Earth?:  10
enter the speed as a fraction of the speed of light:  0.99


it will take the spaceship 48.98979485566356 years to reach it's destination from the perspective of people on the spaceship


results for a planet 10 light-years away with v = 0.99c.

it will take the spaceship 10.1010101010101 years to reach it's destination from the perspective of earth

it will take the spaceship 48.98979485566356 years to reach it's destination from the perspective of people on the spaceship


#### The semi-empirical mass formula: 
In nuclear physics, the semi-empirical mass formula calculates the approximate nuclear binding energy _B_ of an atomic nucleus with atomic number _Z_ and mass number _A_. The formula reads as:


\begin{equation}
B = a_1A - a_2A^{2/3} - a_3\frac{Z^2}{A^{1/3}} - a_4\frac{(A-2Z)^2}{A}+ \frac{a_5}{\sqrt{A}}
\end{equation}

where (in units of millions of electron volts, MeV for short), the constants 

\begin{equation}
a_1 = 15.67
\end{equation}
\begin{equation}
a_2 = 17.23
\end{equation}
\begin{equation}
a_3 = 0.75
\end{equation}
\begin{equation}
a_4 = 93.2 
\end{equation}


and \begin{equation}
a_5 = \text{0 if A is odd \\\\ 12.0  if A and Z are both even, \\\\ -12.0 if A is even and Z is odd}
\end{equation}

Write a program that takes as its input the values of _A_ and _Z_, and prints out 

(a) the binding energy _B_ for the corresponding atom and 

In [70]:
#function that returns binding energy
def binding(A,Z):
    #constants in MeV
    a1 = 15.67
    a2 = 17.23
    a3 = 0.75
    a4 = 93.2
    #if the remainder of A divided by 2 is 1, it is odd, and a5 is 0
    if A%2 == 1:
        a5 = 0
    #if the remainder of A/2 and Z/2 is 0, its even and a5 is 12
    elif A%2 == 0 and Z%2 == 0:
        a5 = 12
    #if the remainder of A/2 is 0 and Z/2 does not equal 0, a5 is -12
    elif A%2 == 0 and Z%2 != 0:
        a5 = -12
    #binding energy equation
    B = (a1 * A  - a2 * A**(2/3) - a3 * (Z**2 / A**(1/3)) - a4 * ((A - 2*Z)**2 / A) + a5 / A**(1/2))
    #print binding energy
    print("the binding energy is", str(B), "MeV")

#ask user for input on the mass and atomic number
A = float(input("what is the mass number, A? "))
Z = int(input("what is the atomic number, Z?: "))

binding(A,Z)

what is the mass number, A?  58
what is the atomic number, Z?:  28


the binding energy is 493.93560680136824 MeV


(b) the binding energy per nucleon, which is _B/A_. 

In [71]:
##function that returns binding energy per nucleon
def bindingpernucleon(A,Z):
    #constants in MeV
    a1 = 15.67
    a2 = 17.23
    a3 = 0.75
    a4 = 93.2
    #if the remainder of A divided by 2 is 1, it is odd, and a5 is 0
    if A%2 == 1:
        a5 = 0
    #if the remainder of A/2 and Z/2 is 0, its even and a5 is 12
    elif A%2 == 0 and Z%2 == 0:
        a5 = 12
    #if the remainder of A/2 is 0 and Z/2 does not equal 0, a5 is -12
    elif A%2 == 0 and Z%2 != 0:
        a5 = -12
    #binding energy equation
    B = (a1*A-a2 * A**(2/3)-a3*(Z**2 / A**(1/3))-a4 * ((A - 2*Z)**2 / A) + a5 / A**(1/2))
    #BdA represents binding energy per nucleon
    BdA = B/A
    #return binding energy
    print("the binding energy per nucleon is", str(BdA), "MeV")

A = float(input("what is the mass number, A? "))
Z = int(input("what is the atomic number, Z?: "))

bindingpernucleon(A,Z)

what is the mass number, A?  58
what is the atomic number, Z?:  28


the binding energy per nucleon is 8.516131151747729 MeV


Use your program to find the binding energy of an atom with _A_ = 58 and _Z_ = 28. 

(Hint: the correct answer is around 490 MeV.)

In [107]:
binding(58,28)

the binding energy is 493.93560680136824 MeV


Use your program to find the binding energy of an atom with A = 58 and Z = 28 \
Answer is the binding energy is 493.93560680136824 MeV